int32 sender_robot_id
bool is_converged
float64 current_grad_norm